brainbeatsback/server.js

```diff
@@ -119,6 +116,194 @@ app.post('/api/resetpassword', function(req, res) {
     });
 })
 
+/*
+    Example: 
+        GET localhost:4000/api/models
+        Headers- Content-Type: application/json; charset=utf-8
+        Response- 200 OK
+    * You MUST supply the exact Content-Type above, or it won't work. 
+*/
+app.get('/api/models', function(req, res) {
+    var conn = getConnection();
+
+    Model.find().then(function(doc) {
+        if (doc == null) {
+            res.status(404).send("No models found.");
+        } else {
+            var data = []
+            doc.forEach(model => data.push(model.name));
+            res.status(200).send(data);
+        }
+    });
+})
+
+/*
+    Example: 
+        GET localhost:4000/api/models/all
+        Headers- Content-Type: application/json; charset=utf-8
+        Response- 200 OK
+    * You MUST supply the exact Content-Type above, or it won't work. 
+*/
+app.get('/api/models/all', function(req, res) {
+    var conn = getConnection();
+
+    Model.find().then(function(doc) {
+        if (doc == null) {
+            res.status(404).send("No models found.");
+        } else {
+            res.status(200).send(doc);
+        }
+    });
+})
+
+/*
+    Example: 
+        GET localhost:4000/api/models/MODEL_NAME
+        Headers- Content-Type: application/json; charset=utf-8
+        Response- 200 OK
+    * You MUST supply the exact Content-Type above, or it won't work. 
+*/
+app.get('/api/models/:model_name', function(req, res) {
+    var conn = getConnection();
+
+    var model_name = req.params.model_name;
+
+    Model.findOne({"name": model_name}).then(function(doc) {
+        if (doc == null) {
+            res.status(404).send("Model does not exist.");
+        } else {
+            res.status(200).send(doc);
+        }
+    });
+})
+
+
+/*
+    Example: 
+        GET localhost:4000/api/midis
+        Headers- Content-Type: application/json; charset=utf-8
+        Body- {"email": "harry@hsauers.net", "password": "Passwd123!"}
+        Response- 200 OK
+    * You MUST supply the exact Content-Type above, or it won't work. 
+    * Note the user's account info in the body. 
+    * 
+    * @TODO - this feels bad. I don't like the various nested levels. fix it at some point. 
+*/
+app.post('/api/midis', async function(req, res) {
+    var body = req.body;
+    var email = body.email;
+    var password = body.password;
+
+    // check credentials
+    User.findOne({"email": email}).then(function(doc) {
```

Comment:
Since it's a relatively small amount of nesting, it should be okay. It's readable and it works. If it had more nesting, then I'd recommend refactoring, but we should be all good!